File: Python/FoundationOfPythonNetworkProgramming/ch03/udpechoserver.rb
#!/usr/bin/env ruby
require 'socket'

host = '127.0.0.1'
port = 51423

socket = UDPSocket.new
socket.bind(host, port)
loop do
  message, address = socket.recvfrom(8192)
  puts "Got message '#{message}' from '#{address[3]}'"
  
  socket.send(message 0, address[3], address[1])
  puts "Sent message to #{address[3]}:#{address[1]}"
end

File: Python/FoundationOfPythonNetworkProgramming/ch03/udpechoserver.py
#!/usr/bin/env python 
# UDP Echo Server - Chapter 3 - udpechoserver.py 
import socket, traceback, time 

host = '127.0.0.1'                               # Bind to all interfaces 
port = 51423 

s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM) 
s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1) 
s.bind((host, port)) 

while 1: 
    try: 
        message, address = s.recvfrom(8192) 
        print "Got message '%s' from %s" % (message, address)

        # Echo it back 
        s.sendto(message, address) 
        print "Sent response to '%s'" % (address,)

    except (KeyboardInterrupt, SystemExit): 
        raise 
    except: 
        traceback.print_exc() 
